fem/intrules.cpp

```diff
@@ -174,6 +174,23 @@ void IntegrationRule::GrundmannMollerSimplexRule(int s, int n)
    }
 }
 
+bool IntegrationPoint::operator==(const IntegrationPoint &ip) const
+{
+   if (x != ip.x) { return false; }
+   if (y != ip.y) { return false; }
+   if (z != ip.z) { return false; }
```

Comment:
Not all of the members {`x`,`y`,`z`} are always initialized, so this comparison may lead to undefined behavior.

I think @rcarson3 was just suggesting to compare the addresses of the `IntegrationRule`s , so we don't need these comparison operators. Comparing the addresses should be sufficient since we use a global container for all the rules and there is no need to create copies of these rules.

Reply:
Oh I see, I misunderstood. Let me push a quick fix.

Reply:
Updated.